pkg/controller/clusterautoscaler/monitoring.go

```diff
@@ -0,0 +1,258 @@
+package clusterautoscaler
+
+import (
+	"context"
+	"fmt"
+
+	monitoringv1 "github.com/coreos/prometheus-operator/pkg/apis/monitoring/v1"
+	autoscalingv1 "github.com/openshift/cluster-autoscaler-operator/pkg/apis/autoscaling/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/equality"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+// ensureAutoscalerServiceMonitor will create a service
+// for the given ClusterAutoscaler custom resource instance.
+func (r *Reconciler) ensureAutoscalerService(ca *autoscalingv1.ClusterAutoscaler) error {
+	desired := r.AutoscalerService(ca)
+	// Set ClusterAutoscaler instance as the owner and controller.
+	if err := controllerutil.SetControllerReference(ca, desired, r.scheme); err != nil {
+		return err
+	}
+
+	current := &corev1.Service{}
+	if err := r.client.Get(context.TODO(), r.AutoscalerName(ca), current); err != nil {
```

Comment:
I think GitHub ate my previous comment because of a force push, so sorry if this shows up twice.

Would it be easier in these functions to just use [CreateOrUpdate](https://godoc.org/sigs.k8s.io/controller-runtime/pkg/controller/controllerutil#CreateOrUpdate)?